middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -194,7 +194,9 @@ def setup_webrick(is_logging)
             http_opts[:SSLPrivateKey] = OpenSSL::PKey::RSA.new File.read ssl_private_key
           else
             # use a generated self-signed cert
-            http_opts[:SSLCertName] = [%w(CN localhost), %W(CN #{host})].uniq
+            cert, key = create_self_signed_cert(1024, [["CN", host]], "Middleman Preview Server")
```

Comment:
Thanks, I had noticed that. I left it as 1024 because accepting self signed cents isn't exactly secure either, and Webrick also hard codes 1024 bits. I wanted to change as little as possible in one go. 

> On May 8, 2015, at 1:35 AM, Dennis Günnewig <notifications@github.com> wrote:
> 
> In middleman-core/lib/middleman-core/preview_server.rb:
> 
> > @@ -194,7 +194,9 @@ def setup_webrick(is_logging)
> >              http_opts[:SSLPrivateKey] = OpenSSL::PKey::RSA.new File.read ssl_private_key
> >            else
> >              # use a generated self-signed cert
> > -            http_opts[:SSLCertName] = [%w(CN localhost), %W(CN #{host})].uniq
> > +            cert, key = create_self_signed_cert(1024, [["CN", host]], "Middleman Preview Server")
> A bit picky, but - although it's local - 1024 is considered to be broken... We should a least use 2048.
> 
> —
> Reply to this email directly or view it on GitHub.
>